src/components/appSwitch/AppSwitch.js

```diff
@@ -37,7 +36,11 @@ class AppSwitch extends React.Component<LoadingProps, {}> {
    */
   constructor(props: LoadingProps) {
     super(props)
-    this.checkAuthStatus()
+    this.ready = false
+    this.checkAuthStatus().then(r => {
+      this.ready = true
+      this.forceUpdate()
```

Comment:
In order to avoid using explictly forceUpdate ready could be a state prop.

Reply:
1. Since it happens in constructor using state is a problem
2. I found out that redux rerenders on every store change not only on items you subscribe to. So we'll need to make different stores for different things.
The question is if "loading" in this case should be effected by external changes, i'm not sure it is required by any other screen at the moment